pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1194,21 +1194,26 @@ func applyUpstreamTLSSettings(opts *buildClusterOpts, tls *networking.TLSSetting
 			ConfigType: &core.TransportSocket_TypedConfig{TypedConfig: util.MessageToAny(tlsContext)},
 		}
 	}
-	// convert to transport socket matcher if the mode was auto detected
-	if tls.Mode == networking.TLSSettings_ISTIO_MUTUAL && mtlsCtxType == autoDetected && util.IsIstioVersionGE14(proxy) {
-		transportSocket := cluster.TransportSocket
-		cluster.TransportSocket = nil
-		cluster.TransportSocketMatches = []*apiv2.Cluster_TransportSocketMatch{
-			{
-				Name: "tlsMode-" + model.IstioMutualTLSModeLabel,
-				Match: &structpb.Struct{
-					Fields: map[string]*structpb.Value{
-						model.TLSModeLabelShortname: {Kind: &structpb.Value_StringValue{StringValue: model.IstioMutualTLSModeLabel}},
+
+	// For `Cluster_ORIGINAL_DST` cluster discovery type, the LbPolicy is `Cluster_CLUSTER_PROVIDED` now
```

Comment:
is there any param to refer cluster_discovery_type directly? will be nicer if we can use that, reduce one level indirection and coupling knowledge...
if not,

could we start with "for headless service, discover type will be original_dst, .and corresponding lbpolicy is CLUSTER_provided".

thus we know the source of motivation is from headless for future developer.

Reply:
make sense